src/views/Transitioner.js

```diff
@@ -172,12 +172,17 @@ class Transitioner extends React.Component<Props, State> {
 
     // update scenes and play the transition
     this._isTransitionRunning = true;
-    this.setState(nextState, () => {
-      nextProps.onTransitionStart &&
-        nextProps.onTransitionStart(
+    this.setState(nextState, async () => {
+      if (nextProps.onTransitionStart) {
+        const result = nextProps.onTransitionStart(
           this._transitionProps,
           this._prevTransitionProps
         );
+
+        if (result instanceof Promise) {
+          await result;
```

Comment:
Wrap in a try/catch blog? We should handle a rejected promise (currently we get a warning)

![simulator screen shot - iphone 8 - 2017-11-28 at 11 37 57](https://user-images.githubusercontent.com/1105429/33334776-a5badfb8-d430-11e7-941b-205c5b0b2d4c.png)